### PR TITLE
Implementation of the location request

### PR DESCRIPTION
**Pull request by Juliana-Dimeglio**
Implementation of the android plugin 

https://user-images.githubusercontent.com/85245831/132056976-dac4bcb3-cb9f-4763-90b6-178e235e51a2.mp4

### CODE REVIEW

File: android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt [hunk 1]
```diff
@@ -26,46 +35,103 @@ class AndroidPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
...
+        object : EventChannel.StreamHandler {
+            override fun onListen(arguments: Any?, events: EventChannel.EventSink) {
+                locationEventSource = events
+            }
+
+            override fun onCancel(arguments: Any?) {
+
+            }
+        }
+
```
Comment (MarcosCGlobant) [inline android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt:56]: remove this extra line please

File: example/lib/main.dart [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import 'package:android_plugin/android_plugin.dart';
+import 'package:android_plugin_example/utils/numeric_constants.dart';
```
Comment (MarcosCGlobant) [inline example/lib/main.dart]: check if you can use relative imports here

File: android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt [hunk 3]
```diff
@@ -96,18 +164,27 @@ class AndroidPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
...
         const val CHANNEL = "android_plugin"
+        const val GET_LOCATION = "getLocation"
+        const val INITIALIZE = "initialize"
+        const val STOP_LOCATION_REQUEST = "stopLocationRequest"
+        const val INTERVAL: Long = 5000
+        const val LOCATION_STOPPED = "Location request stopped"
+        const val ERROR = "Error"
+        const val PERMISSION_DENIED = "Permission denied"
+        const val LOCATION_CHANNEL = "location_channel"
+        const val LONGITUDE_TAG = "Longitude:"
```
Comment (FranciscoGaraventa) [inline android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt]: I think you missed a white space here after ':'

File: example/lib/main.dart [hunk 4]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
-            ElevatedButton(
-              onPressed: () {
-                requestPermission();
-              },
-              child: Text(
-                StringConstants.requestPermissionButton,
+          ),
+          child: Center(
+            child: Container(
+              margin: new EdgeInsets.only(
```
Comment (MarcosCGlobant) [inline example/lib/main.dart]: you can remove the "new" keyword, it's not necessary

File: example/lib/main.dart [hunk 5]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
+                top: NumericConstants.marginTop,
+              ),
+              child: Column(
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                    children: [
+                      Container(
+                        margin: new EdgeInsets.all(
```
Comment (MarcosCGlobant) [inline example/lib/main.dart]: remove "new" here too

File: example/lib/main.dart [hunk 6]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
+                                      fontSize: NumericConstants
+                                          .coordinatesTextFontSize,
+                                    ),
+                                  ),
+                                  Text(
+                                    snapshot.data.toString().split("%").last,
+                                    style: TextStyle(
+                                      fontSize: NumericConstants
+                                          .coordinatesTextFontSize,
+                                    ),
```
Comment (FranciscoGaraventa) [inline example/lib/main.dart]: When you pressed on stop location it's shown two times the same message, can you try yo show only one 'Location Request Stoped'?

File: example/lib/main.dart [hunk 7]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
+                        ),
+                      ),
+                    ],
+                  ),
+                  StreamBuilder(
+                    stream: AndroidPlugin.locationStream,
+                    builder: (context, snapshot) {
+                      return snapshot.hasData
+                          ? Container(
+                              padding: new EdgeInsets.all(
```
Comment (MarcosCGlobant) [inline example/lib/main.dart]: remove "new" here too

File: example/lib/main.dart [hunk 8]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
+                        height: NumericConstants.heightGestureDetector,
+                        decoration: BoxDecoration(
+                          borderRadius: BorderRadius.circular(
+                            NumericConstants.borderRadiusGestureDetector,
+                          ),
+                          color: Color.fromRGBO(
+                            191,
+                            6,
+                            3,
+                            2,
```
Comment (FranciscoGaraventa) [inline example/lib/main.dart]: This can be added to a constant file

File: example/lib/main.dart [hunk 9]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
-                  StringConstants.checkPermissionButton,
-                ),
-              ),
+        body: Container(
+          decoration: BoxDecoration(
+            color: Color.fromRGBO(
+              178,
+              176,
+              155,
+              2,
```
Comment (FranciscoGaraventa) [inline example/lib/main.dart]: This can be added to a constant file

File: example/lib/main.dart [hunk 10]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
     return MaterialApp(
       home: Scaffold(
         appBar: AppBar(
+          backgroundColor: Color.fromRGBO(
+            141,
+            8,
+            1,
+            2,
```
Comment (FranciscoGaraventa) [inline example/lib/main.dart]: This can be added to a constant file

File: example/lib/main.dart [hunk 11]
```diff
@@ -48,38 +87,213 @@ class _MyAppState extends State<MyApp> {
...
+          ),
```
Comment (MarcosCGlobant) [inline example/lib/main.dart:95]: you can move this color (and the other ones below) to a constants file and make the widget more readable

**Review by MarcosCGlobant — COMMENTED** [review]
minor styles comments

File: example/lib/main.dart (hunk 10 above)
Comment (luisjavierglb) [reply example/lib/main.dart]: 👍

**Review by luisjavierglb — APPROVED** [review]
LGTM

File: android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt (hunk 3 above)
Comment (Juliana-Dimeglio) [reply android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt]: Fixed

File: example/lib/main.dart (hunk 8 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart (hunk 6 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart (hunk 9 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart (hunk 10 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt (hunk 1 above)
Comment (Juliana-Dimeglio) [reply android/src/main/kotlin/com/example/android_plugin/AndroidPlugin.kt:56]: Fixed

File: example/lib/main.dart (hunk 2 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart (hunk 4 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart (hunk 7 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart (hunk 5 above)
Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

File: example/lib/main.dart [hunk 12]
```diff
@@ -14,72 +17,311 @@ class MyApp extends StatefulWidget {
...
+                                          style: TextStyle(
+                                            fontSize: NumericConstants
+                                                .coordinatesTextFontSize,
+                                          ),
+                                        ),
+                                      ],
+                                    )
+                                  : Text(
+                                      snapshot.data.toString(),
+                                    ),
```
Comment (MarcosCGlobant) [inline example/lib/main.dart]: You should use the same style for this Text, so it keeps consistency between messages

Comment (Juliana-Dimeglio) [reply example/lib/main.dart]: Fixed

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Pull request merged**